### PR TITLE
FUSETOOLS2-54 - support using different Camel Catalog version

### PR DESCRIPTION
**Pull request by apupier**
Signed-off-by: Aurélien Pupier <apupier@redhat.com>

# Pull Request informations

## Rebase & Merge default requirements

1. Green build for master branch
2. Wait 24 hours after PR creation
3. Green job for PR
4. Approved PR

## PR labels default process

- READY_FOR_REVIEW  &rightarrow;  REVIEW_DONE  &rightarrow;  READY_FOR_MERGE

## Tests

- [ ] Are there **Unit tests**?
- [ ] Are there **Integration tests**?
- [ ] Do we need a new **UI test**?

## PR workflow progress

1. [ ] Tagged with relevant **PR labels**
2. [ ] Green **job for PR**
3. [ ] PR was created more than **24 hours ago** or **All committers approved** it
4. [ ] Green **master** branch build

### CODE REVIEW

**Comment by apupier** [comment]
to more easily review this PR, it is possible to play with the feature in VS Code with this other PR https://github.com/camel-tooling/camel-lsp-client-vscode/pull/182

The result is something like:
CatalogVersionPreference

**Comment by lhein** [comment]
What happens if the catalog version is not existing at all or has to be downloaded?

**Comment by apupier** [comment]
> What happens if the catalog version is not existing at all

there is a log
https://github.com/camel-tooling/camel-language-server/pull/324/files#diff-6155805cf1e1e6f08ce96bdff48db650R94

It falls back to the embedded version.

> or has to be downloaded?

it downloads it using the Camel catalog capabilities

https://github.com/camel-tooling/camel-language-server/pull/324/files#diff-6155805cf1e1e6f08ce96bdff48db650R92

**Comment by lhein** [comment]
Ok, so it will take some time for the download and there might be a delay until you get your completion?

**Comment by apupier** [comment]
> Ok, so it will take some time for the download and there might be a delay until you get your completion?

yes. there is no easy way to indicate it from Language server side. I mentioned it on client side https://github.com/camel-tooling/camel-lsp-client-vscode/pull/182/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R69

File: src/test/java/com/github/cameltooling/lsp/internal/CamelCatalogVersionTest.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	
+	@Test
+	public void testCompletionFallbackWithInvalidVersion() throws Exception {
+		camelCatalogVersion = "invalid";
+		
+		CamelLanguageServer camelLanguageServer = basicCompletionCheck();
+		
+		assertThat(camelLanguageServer.getTextDocumentService().getCamelCatalog().get().getLoadedVersion()).isNull();
+	}
+
```
Comment (lhein) [inline src/test/java/com/github/cameltooling/lsp/internal/CamelCatalogVersionTest.java]: how about adding a null test too?

**Pull request merged**